### PR TITLE
Code and related files for the dependency injection demos.

### PR DESCRIPTION
**Pull request by reymalahay**
Hi Eugen,

This is just to let you know that my work is ready for review.

Kind regards,
rey malahay

### CODE REVIEW

**Comment by eugenp** [comment]
Hey @reymalahay - jumping in here with a quick note (unrelated to the review) - I noticed the PR includes a lot of metadata files - let's get rid of these entirely. The goal is to keep things very clean and minimal in terms of changes - both to make the review easy as well as well as the work focused on what's actually going on in the article. 
Cheers, 
Eugen.

**Comment by reymalahay** [comment]
Hi @eugenp 

I removed the metadata files and pushed to the server.

Kind regards,
rey malahay

**Comment by eugenp** [comment]
@reymalahay Sounds good, but looking at the PR I still see 103 files changed - so it looks like either the PR isn't updated, or we still have a few files that we should remove from it. 
Cheers, 
Eugen.

**Comment by reymalahay** [comment]
@eugenp i've removed the .metadata files from my commit.

**Comment by eugenp** [comment]
@reymalahay Down to 42 files? Are we really changing 42 actual files for the article?

**Comment by reymalahay** [comment]
@eugenp i excluded a lot of files that belong to the .metadata directory. those were generated by eclipse and not related to my contribution.

**Comment by eugenp** [comment]
@reymalahay - yes, that's what I'm thinking as well, that most of these files are actually not related to the actual logic you're working on. So - everything that's unrelated needs to be excluded. 
Basically - the PR should literally only have the changes that are actually part of the logic you're implementing. 
Hope that makes sense. 
Cheers, 
Eugen.

**Comment by reymalahay** [comment]
@eugenp it makes sense. those metadata files were included when i checked out the code. i got the impression that i should leave as is. anyway, lesson learned.

**Comment by eugenp** [comment]
@reymalahay So, the metadata files are generated by your IDE locally - that's why you see them as part of the PR. 
The way to figure that out is - when you open the PR - have a look at the files that are in it. Usually you should only see 2-3 files in there. 
OK, so, when you remove these, let me now and I'll have a look. 
Cheers.

**Comment by reymalahay** [comment]
@eugenp I removed the metadata files from my commit and pushed to the remote repository.

**Comment by eugenp** [comment]
Hey Rey - just to make sure I'm not seeing something different - how many files are you seeing as part of this PR?

**Comment by reymalahay** [comment]
@eugenp there are 42 files. aside from the .gitignore file, all the files are new.

**Comment by eugenp** [comment]
@reymalahay OK, so we're seeing the same thing then. 
My question is - why does the PR have 42 files? 
I was thinking it should have 2-3 files, definitely not as many as 42. 
As far as I see you're still including a lot of extra, metadata, generated files.

**Comment by reymalahay** [comment]
@eugenp i did a directory listing of my commit:

├── build.gradle
├── docs
│   ├── autowired-name-demo-classdiagram.dia
│   ├── autowired-name-demo-classdiagram.dia~
│   ├── autowired-name-demo-classdiagram.png
│   ├── autowired-type-demo-classdiagram.dia
│   ├── autowired-type-demo-classdiagram.dia~
│   ├── autowired-type-demo-classdiagram.png
│   ├── inject-demo-classdiagram.dia
│   ├── inject-demo-classdiagram.dia~
│   ├── inject-demo-classdiagram.png
│   ├── inject-name-demo-classdiagram.dia
│   ├── inject-name-demo-classdiagram.dia~
│   ├── inject-name-demo-classdiagram.png
│   ├── inject-qualifier-demo-classdiagram.dia
│   ├── inject-qualifier-demo-classdiagram.dia~
│   ├── inject-qualifier-demo-classdiagram.png
│   ├── resource-demo-classdiagram.dia
│   ├── resource-demo-classdiagram.dia~
│   ├── resource-demo-classdiagram.png
│   ├── resource-demo-usecase.dia
│   ├── resource-demo-usecase.dia~
│   ├── resource-method-byname-demo-classdiagram.dia
│   ├── resource-method-byname-demo-classdiagram.dia~
│   └── resource-method-byname-demo-classdiagram.png
└── src
    └── test
        ├── java
        │   └── com
        │       └── baeldung
        │           ├── autowired
        │           │   ├── FieldAutowiredDemo.java
        │           │   ├── FieldAutowiredNameDemo.java
        │           │   └── FieldQualifierAutowiredDemo.java
        │           ├── dependency
        │           │   ├── AnotherArbitraryDependency.java
        │           │   ├── ArbitraryDependency.java
        │           │   └── YetAnotherArbitraryDependency.java
        │           ├── inject
        │           │   ├── FieldByNameInjectDemo.java
        │           │   ├── FieldInjectDemo.java
        │           │   └── FieldQualifierInjectDemo.java
        │           └── resource
        │               ├── FieldResourceInjectionDemo.java
        │               ├── MethodByQualifierResourceDemo.java
        │               ├── MethodByTypeResourceDemo.java
        │               ├── MethodResourceInjectionDemo.java
        │               ├── NamedResourceTest.java
        │               ├── QualifierResourceInjectionDemo.java
        │               └── SetterResourceInjectionDemo.java
        └── resources
            ├── applicationContextTest-@Autowired-Name.xml
            ├── applicationContextTest-@Autowired-Qualifier.xml
            ├── applicationContextTest-@Autowired-Type.xml
            ├── applicationContextTest-@Inject-Name.xml
            ├── applicationContextTest-@Inject-Qualifier.xml
            ├── applicationContextTest-@Inject-Type.xml
            ├── applicationContextTest-@Resource-NameType.xml
            └── applicationContextTest-@Resource-Qualifier.xml

basically, it includes the image files, integration test class files, xml files. All of these files are the code demos that are used in the article.

**Comment by eugenp** [comment]
@reymalahay Sounds good. 
So - here's a few of the files you can probably remove. 
All the files ending with ~
build.gradle (all of the projects in the repo are using Maven, so unless Gradle is the topic of the article - it's better to use Maven)
I'm not sure we need the .dia files - the pngs should be enough
Also note that most articles on the site don't make a lot of use of images, so if you can, show the concepts via code only. 

Cheers, 
Eugen.

**Comment by reymalahay** [comment]
@eugenp, the dia files are the source files used to generate the png's. dia is an open source diagram tool, an alternative to visio, http://dia-installer.de/doc/en/index.html. i'll make the necessary modifications to change the build tool to maven (i moved to gradle many years ago)

**Comment by eugenp** [comment]
Sounds good.

**Comment by reymalahay** [comment]
@eugenp I added the pom.xml file so that the demo code is accessible to Maven user. i left the build.gradle file.

**Comment by reymalahay** [comment]
@eugenp I added the pom.xml file so that the demo code is accessible to Maven users. i left the build.gradle file.

**Comment by eugenp** [comment]
@reymalahay Sounds good

**Comment by reymalahay** [comment]
@eugenp i pushed revisions to the code as per discussion with Alex Theedom
